ctk/common/src/test/kotlin/org/codice/compilance/verification/core/ResponseVerifierSpec.kt

```diff
@@ -0,0 +1,206 @@
+                ResponseVerifierTest(request, it, HTTP_POST).verify()
+            }
+        }
+
+        "response with non-unique ID should fail" {
+            Common.buildDom(createResponse(id = "not-unique-id")).let {
+                ResponseVerifierTest(request, it, HTTP_POST).verify()
+            }
+
+            Common.buildDom(createResponse(id = "not-unique-id")).let {
+                shouldThrow<SAMLComplianceException> {
+                    ResponseVerifierTest(request, it, HTTP_POST).verify()
+                }.message?.apply {
+                    shouldContain(SAMLCore_1_3_4_a.message)
+                    shouldContain(SAMLCore_3_2_2_a.message)
+                }
+            }
+        }
+
+        "response with incorrect InResponseTo should fail" {
+            Common.buildDom(createResponse(inResponseTo = "incorrect")).let {
+                shouldThrow<SAMLComplianceException> {
+                    ResponseVerifierTest(request, it, HTTP_POST).verify()
+                }.message?.shouldContain(SAMLCore_3_2_2_b.message)
+            }
+        }
+
+        "response with blank version should fail" {
+            Common.buildDom(createResponse(version = " ")).let {
+                shouldThrow<SAMLComplianceException> {
+                    ResponseVerifierTest(request, it, HTTP_POST).verify()
+                }.message?.shouldContain(SAMLCore_1_3_1_a.message)
+            }
+        }
+
+        "response with non-utc instant issuer should fail" {
+            Common.buildDom(createResponse(instant = "2018-05-01T06:15:30-07:00")).let {
+                shouldThrow<SAMLComplianceException> {
+                    ResponseVerifierTest(request, it, HTTP_POST).verify()
+                }.message?.apply {
+                    this.shouldContain(SAMLCore_1_3_3_a.message)
+                    this.shouldContain(SAMLCore_3_2_2_d.message)
+                }
+            }
+        }
+
+        "response with correct destination should pass" {
+            Common.buildDom(createResponse(attribute = "$DESTINATION=\"$correctUri\"")).let {
+                ResponseVerifierTest(request, it, HTTP_POST).verify()
+            }
+        }
+
+        "response with incorrect destination should fail" {
+            Common.buildDom(createResponse(attribute = "$DESTINATION=\"$incorrectUri\"")).let {
+                shouldThrow<SAMLComplianceException> {
+                    ResponseVerifierTest(request, it, HTTP_POST).verify()
+                }.message?.shouldContain(SAMLCore_3_2_2_e.message)
+            }
+        }
+
+        "response with correct consent should pass" {
+            Common.buildDom(createResponse(attribute = "$CONSENT=\"$correctUri\"")).let {
+                ResponseVerifierTest(request, it, HTTP_POST).verify()
+            }
+        }
+
+        "response with non-uri consent should fail" {
+            Common.buildDom(createResponse(attribute = "$CONSENT=\"$incorrectUri\"")).let {
```

Comment:
❓ I assume that this test right here is what found our bug above? 👍

Reply:
Yes